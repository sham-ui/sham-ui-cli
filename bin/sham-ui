#!/usr/bin/env node

require( 'commander' )
    .version( require( '../package' ).version )
    .usage( '<command> [options]' )
    .command( 'new', 'generate a new project from a template' ).alias( 'n' )
    .command( 'upgrade', 'upgrade existed project from a template' ).alias( 'u' )
    .command( 'generate', 'generate a new component' ).alias( 'g' )
    .command( 'collect', 'collect filters & directives' ).alias( 'c' )
    .command( 'destroy', 'destroy component' ).alias( 'd' )
    .parse( process.argv );