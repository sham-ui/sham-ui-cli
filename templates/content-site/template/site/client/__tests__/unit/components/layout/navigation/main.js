import { createDI } from 'sham-ui';
import * as directives from 'sham-ui-directives';
import hrefto from 'sham-ui-router/lib/href-to';
import LayoutNavigationMain  from '../../../../../src/components/layout/navigation/main.sfc';
import renderer from 'sham-ui-test-helpers';

it( 'renders correctly', () => {
    const DI = createDI();
    DI.bind( 'router', {
        storage: {
            params: {
                page: 2
            },
            url: '',
            addWatcher() {}
        },
        generate: () => '/'
    } );

    const meta = renderer( LayoutNavigationMain, {}, {
        DI,
        directives: {
            ...directives,
            hrefto
        }
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );


it( 'toggle navbar', () => {
    const DI = createDI();
    DI.bind( 'router', {
        storage: {
            params: {
                page: 2
            },
            url: '',
            addWatcher() {}
        },
        generate: () => '/'
    } );

    const meta = renderer( LayoutNavigationMain, {}, {
        DI,
        directives: {
            ...directives,
            hrefto
        }
    } );
    meta.ctx.container.querySelector( '.navbar-collapse' ).click();
    expect( meta.toJSON() ).toMatchSnapshot();
    meta.ctx.container.querySelector( '.navbar-collapse' ).click();
    expect( meta.toJSON() ).toMatchSnapshot();
} );
