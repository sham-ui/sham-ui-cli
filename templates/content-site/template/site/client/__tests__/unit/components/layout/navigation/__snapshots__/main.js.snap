// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "Options": Object {
    "onOpenSearchForm": [Function],
  },
  "html": "
<nav class=\\"navbar navbar-expand-lg fixed-top\\">
  <div class=\\"container-fluid\\">
    <div class=\\"logo\\">
      <a href=\\"/\\">
        <img src=\\"/images/logo-dark.png\\" alt=\\"logo\\">
      </a>
    </div>
    <!--0-->
    <div class=\\"collapse navbar-collapse\\">
      <ul class=\\"navbar-nav ml-auto mr-auto\\">
        <li class=\\"nav-item\\">
          <a class=\\"nav-link \\" href=\\"/\\">Home</a>
        </li>
        <li class=\\"nav-item\\">
          <a class=\\"nav-link active\\" href=\\"/\\">Contact</a>
        </li>
      </ul>
    </div>
    <div class=\\"navbar-right ml-auto\\">
      <div class=\\"theme-switch-wrapper\\">
        <label class=\\"theme-switch\\" for=\\"checkbox\\">
          <input type=\\"checkbox\\" id=\\"checkbox\\">
          <div class=\\"slider round\\"></div>
        </label>
      </div>
      <!--1-->
      <div class=\\"search-icon\\">
        <i class=\\"icon-search\\"></i>
      </div>
      <button class=\\"navbar-toggler\\" type=\\"button\\" aria-expanded=\\"false\\" aria-label=\\"Toggle navigation\\"><span class=\\"navbar-toggler-icon\\"></span></button>
    </div>
  </div>
</nav>
",
}
`;

exports[`toggle navbar 1`] = `
Object {
  "Options": Object {
    "onOpenSearchForm": [Function],
  },
  "html": "
<nav class=\\"navbar navbar-expand-lg fixed-top\\">
  <div class=\\"container-fluid\\">
    <div class=\\"logo\\">
      <a href=\\"/\\">
        <img src=\\"/images/logo-dark.png\\" alt=\\"logo\\">
      </a>
    </div>
    <!--0-->
    <div class=\\"collapse navbar-collapse show\\" style=\\"max-height: 0px;\\">
      <ul class=\\"navbar-nav ml-auto mr-auto\\">
        <li class=\\"nav-item\\">
          <a class=\\"nav-link \\" href=\\"/\\">Home</a>
        </li>
        <li class=\\"nav-item\\">
          <a class=\\"nav-link active\\" href=\\"/\\">Contact</a>
        </li>
      </ul>
    </div>
    <div class=\\"navbar-right ml-auto\\">
      <div class=\\"theme-switch-wrapper\\">
        <label class=\\"theme-switch\\" for=\\"checkbox\\">
          <input type=\\"checkbox\\" id=\\"checkbox\\">
          <div class=\\"slider round\\"></div>
        </label>
      </div>
      <!--1-->
      <div class=\\"search-icon\\">
        <i class=\\"icon-search\\"></i>
      </div>
      <button class=\\"navbar-toggler\\" type=\\"button\\" aria-expanded=\\"false\\" aria-label=\\"Toggle navigation\\"><span class=\\"navbar-toggler-icon\\"></span></button>
    </div>
  </div>
</nav>
",
}
`;

exports[`toggle navbar 2`] = `
Object {
  "Options": Object {
    "onOpenSearchForm": [Function],
  },
  "html": "
<nav class=\\"navbar navbar-expand-lg fixed-top\\">
  <div class=\\"container-fluid\\">
    <div class=\\"logo\\">
      <a href=\\"/\\">
        <img src=\\"/images/logo-dark.png\\" alt=\\"logo\\">
      </a>
    </div>
    <!--0-->
    <div class=\\"collapse navbar-collapse\\" style=\\"\\">
      <ul class=\\"navbar-nav ml-auto mr-auto\\">
        <li class=\\"nav-item\\">
          <a class=\\"nav-link \\" href=\\"/\\">Home</a>
        </li>
        <li class=\\"nav-item\\">
          <a class=\\"nav-link active\\" href=\\"/\\">Contact</a>
        </li>
      </ul>
    </div>
    <div class=\\"navbar-right ml-auto\\">
      <div class=\\"theme-switch-wrapper\\">
        <label class=\\"theme-switch\\" for=\\"checkbox\\">
          <input type=\\"checkbox\\" id=\\"checkbox\\">
          <div class=\\"slider round\\"></div>
        </label>
      </div>
      <!--1-->
      <div class=\\"search-icon\\">
        <i class=\\"icon-search\\"></i>
      </div>
      <button class=\\"navbar-toggler\\" type=\\"button\\" aria-expanded=\\"false\\" aria-label=\\"Toggle navigation\\"><span class=\\"navbar-toggler-icon\\"></span></button>
    </div>
  </div>
</nav>
",
}
`;
